app/scripts/lib/security-provider-helpers.ts

```diff
@@ -1,35 +1,46 @@
+import { Json } from '@metamask/utils';
+import { MessageParams } from '@metamask/message-manager';
 import getFetchWithTimeout from '../../../shared/modules/fetch-with-timeout';
 import { MESSAGE_TYPE } from '../../../shared/constants/app';
 
 const fetchWithTimeout = getFetchWithTimeout();
 
+export interface RequestData {
+  origin: string;
+  msgParams?: MessageParams;
+  messageParams?: MessageParams;
+  txParams?: Record<string, unknown>;
+}
+
 export async function securityProviderCheck(
-  requestData,
-  methodName,
-  chainId,
-  currentLocale,
-) {
+  requestData: RequestData,
+  methodName: string,
+  chainId: string,
+  currentLocale: string,
+): Promise<Record<string, Json>> {
   let dataToValidate;
+  // Core message managers use messageParams but frontend uses msgParams with lots of references
+  const params = requestData.msgParams || requestData.messageParams;
```

Comment:
This could have been the concern of the `AbstractMessageManager` as it is currently the edge case with `messageParams`, but I guess it makes sense to support this as low as possible to support other future code paths.